File: 2-Regression/3-Polynomial_Regression/src/regression_template.py
# Polynomial Linear Regression

# Import the Libraries
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd
import statsmodels.formula.api as sm
import os

from sklearn.model_selection import train_test_split
from sklearn.preprocessing import StandardScaler, PolynomialFeatures
from sklearn.linear_model import LinearRegression
from sklearn.preprocessing import LabelEncoder, OneHotEncoder

# Import the data
Path = os.path.abspath(os.path.join(os.path.dirname( __file__ ), '..', 'bin\\Position_Salaries.csv'))
dataset = pd.read_csv(Path)

X = dataset.iloc[:, 1:2].values
Y = dataset.iloc[:, 2].values

# Fitting Regressoin to the data
### Create the regressor here

# Predicting a new result with Non-Linear Regression
Y_pred_ = regressor.predict([[6.5]])

# Visualising the Regression results
plt.scatter(X, Y, color='red')
plt.plot(X, regressor.predict(X), color='blue')
plt.plot(6.5, Y_pred, color = 'green')
plt.title('Truth or Bluff Salary Tester (Regression Model)')
plt.xlabel('Position Level')
plt.ylabel('Salary')
plt.show()

# Visualising the Regression results (for higher resolution)
X_grid = np.arrange(min(X), max(X), 0.1)
X_grid = X_grid.reshape((len(X_grid), 1))
plt.scatter(X, Y, color='red')
plt.plot(X_grid, regressor.predict(X_grid), color='blue')
plt.plot(6.5, Y_pred, color = 'green')
plt.title('Truth or Bluff Salary Tester (Regression Model)')
plt.xlabel('Position Level')
plt.ylabel('Salary')
plt.show()

File: 2-Regression/3-Polynomial_Regression/src/regression_template.R
# Regression Template

# Path to the file and Libraries
Path = 'MachineLearning-Portfolio/2-Regression/3-Polynomial_Regression/bin/Position_Salaries.csv'
Path2 = 'MachineLearning-Portfolio/2-Regression/3-Polynomial_Regression/bin'
library(caTools, help, pos = 2, lib.loc = NULL)
library(ggplot2)
library(lme4)

# Import dataset
dataset = read.csv(Path)
dataset = dataset[2:3]

# Splitting into Training and Test set
#set.seed(123)
#split = sample.split(dataset$Profit, SplitRatio = 0.8)
#training_set = subset(dataset, split == TRUE)
#test_set = subset(dataset, split == FALSE)

# Feature Scaling

# Fitting Poynomial Regression to the dataset
###Create your regression model
poly_reg = lm(formula = Salary ~.,
            data = dataset)

# Predict a new result with Regression
###Edit these values
Y_pred = predict(regressor, data.frame(Level = 6.5))
print(Y_pred)

# Visualising the Regression result
X_grid = seq(min(dataset$Level), max(datase$Level), 0.1)
LinearPlot <- ggplot() +
    geom_point(aes(x = dataset$Level, y = dataset$Salary),
		color = 'red') +
	geom_line(aes(x = dataset$Level, y = predict(lin_reg, newdata = data.frame(Level = X_grid))),
		color = 'blue') +
	ggtitle('Truth of Bluff (Linear Regression)') +
	xlab('Level') +
	ylab('Salary')

ggsave("LinearPlot.png",  path = Path2)